DllImportGenerator/DllImportGenerator/Marshalling/ConditionalStackallocMarshallingGenerator.cs

```diff
@@ -0,0 +1,220 @@
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.Interop
+{
+    internal abstract class ConditionalStackallocMarshallingGenerator : IMarshallingGenerator
+    {
+        private static string GetAllocationMarkerIdentifier(string managedIdentifier) => $"{managedIdentifier}__allocated";
+
+        private static string GetByteLengthIdentifier(string managedIdentifier) => $"{managedIdentifier}__bytelen";
+
+        private static string GetStackAllocIdentifier(string managedIdentifier) => $"{managedIdentifier}__stackptr";
+
+        protected IEnumerable<StatementSyntax> GenerateConditionalAllocationSyntax(
+            TypePositionInfo info, 
+            StubCodeContext context,
+            int stackallocMaxSize,
+            bool checkForNull)
```

Comment:
All current usages seem to set this to true - when would we not want the null check?